File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day03PerfectlySphericalHousesInAVacuumTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day03PerfectlySphericalHousesInAVacuumTest extends FlatSpec with Matchers {
  "2015 Day 3 - Input" should "calculate answers" in {
    val dp = new Day03PerfectlySpericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-input.txt")

    dp.part1Answer should be(2572)
    dp.part2Answer should be(2631)
  }

  "2015 Day 3 - Example #1" should "calculate answers" in {
    val dp = new Day03PerfectlySpericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#1.txt")

    dp.part1Answer should be(2)
    dp.part2Answer should be(3)
  }

  "2015 Day 3 - Example #2" should "calculate answers" in {
    val dp = new Day03PerfectlySpericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#2.txt")

    dp.part1Answer should be(4)
    dp.part2Answer should be(3)
  }

  "2015 Day 3 - Example #3" should "calculate answers" in {
    val dp = new Day03PerfectlySpericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#3.txt")

    dp.part1Answer should be(2)
    dp.part2Answer should be(11)
  }
}

File: java-advent-of-code/src/test/java/advent/twenty_fifteen/Day03PerfectlySphericalHousesInAVacuumTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day03PerfectlySphericalHousesInAVacuumTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day03PerfectlySphericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-input.txt");

        assertEquals(2572, dp.getPart1Answer().intValue());
        assertEquals(2631, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day03PerfectlySphericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#1.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day03PerfectlySphericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#2.txt");

        assertEquals(4, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample3() {
        DailyProblem<Integer, Integer> dp = new Day03PerfectlySphericalHousesInAVacuum("/twenty_fifteen/Day03-PerfectlySphericalHousesInAVacuum-example#3.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(11, dp.getPart2Answer().intValue());
    }
}